cowbird/handlers/handler_factory.py

```diff
@@ -38,23 +45,95 @@ def __init__(self):
         LOGGER.info("Handlers config : [%s]", ", ".join([f"{name} [{cfg.get('active', False)}]"
                                                          for name, cfg in self.handlers_cfg.items()]))
 
+    @overload
+    def create_handler(self, name):
+        # type: (Literal["Catalog"]) -> CatalogHandler
+        ...
+
+    @overload
+    def create_handler(self, name):
+        # type: (Literal["FileSystem"]) -> FileSystemHandler
+        ...
+
+    @overload
+    def create_handler(self, name):
+        # type: (Literal["Geoserver"]) -> GeoserverHandler
+        ...
+
+    @overload
+    def create_handler(self, name):
+        # type: (Literal["Magpie"]) -> MagpieHandler
+        ...
+
+    @overload
+    def create_handler(self, name):
+        # type: (Literal["Nginx"]) -> NginxHandler
+        ...
+
+    @overload
+    def create_handler(self, name):
+        # type: (Literal["Thredds"]) -> ThreddsHandler
+        ...
+
+    @overload
+    def create_handler(self, name):
+        # type: (str) -> Optional[Handler]
+        ...
+
     def create_handler(self, name):
-        # type: (HandlerFactory, str) -> Handler
+        # type: (str) -> Optional[Handler]
```

Comment:
What is the benefit of using all those overload functions above (same question for `get_handler` below)?
Is it because simply using a list of the Literals as input here and a list of Handlers as output doesn't necessarily highlight the relationship between them (ex.: if the input is a `Literal["Thredds"]` it has to return explicitly a `ThreddsHandler` and not any other Handler)?

Reply:
In terms of code, no change, but in the IDE there are better completions because specific types are known. So, if `"Thredds"` is passed as input, `ThreddsHandler` is expected as output instead of `Any(Handler)`. When specific handlers provide extra methods, it will show or not accordingly.